src/core/QCheck2.mli

```diff
@@ -0,0 +1,1992 @@
+           to help you find the problem (called "shrinking")}
+      }
+
+    {[
+      let test = QCheck2.(
+          Test.make
+            ~name:"All lists are sorted"
+            ~count:10_000
+            (list small_nat)
+            (fun l -> l = List.sort compare l));;
+
+      QCheck2.Test.check_exn test;;
+
+      Exception:
+        test `All lists are sorted` failed on ≥ 1 cases:
+        [1; 0] (after 5 shrink steps)
+    ]}
+
+
+    - Generate 20 random trees using {! Gen.fix} :
+
+    {[
+      type tree = Leaf of int | Node of tree * tree
+
+      let leaf x = Leaf x
+      let node x y = Node (x,y)
+
+      let tree_gen = QCheck2.Gen.(sized @@ fix
+                                    (fun self n -> match n with
+                                       | 0 -> map leaf nat
+                                       | n ->
+                                         frequency
+                                           [1, map leaf nat;
+                                            2, map2 node (self (n/2)) (self (n/2))]
+                                    ));;
+
+      QCheck2.Gen.generate ~n:20 tree_gen;;
+    ]}
+
+*)
+
+(** A tree represents a generated value and its successive shrunk values. *)
+module Tree : sig
+  (** Conceptually a pseudo-randomly generated value is packaged with its shrunk values.
+      This coupling - called "integrated shrinking" - in a single type has a major benefit:
+      most generators get shrinking "for free" by composing from smaller generators, and shrinking
+      does not break invariants (e.g. shrinks of a positive number are always positive).
+  *)
+
+  type 'a t
+  (** A tree of random generated values, where the root contains the value used for the test,
+      and the sub-trees contain shrunk values (as trees, to be able to shrink several times a value)
+      used if the test fails. *)
+
+  val root : 'a t -> 'a
+  (** [root tree] returns the root value of the tree of generated values [t]. *)
+
+  val children : 'a t -> 'a t Seq.t
+  (** [children tree] returns the direct sub-trees of the tree of generated values [t]. *)
+
+  val pp : ?depth : int -> (Format.formatter -> 'a -> unit) -> Format.formatter -> 'a t -> unit
+  (** [pp ?depth pp_a ppf tree] pretty-prints the tree of generated values [tree] using the
+      pretty-print formatter [ppf]. Values of type ['a] will be printed using the given
+      pretty-printer [pp_a].
+
+      As a tree [t] can be potentially huge when fully evaluated, you can control the maximum
+      depth the printer goes with [depth].
+      - [None] means "everything"
+      - [0] means "only the root"
+      - [1] means "the root and its direct shrinks"
+      - [2] means "the root, its direct shrinks, and the shrinks of its shrinks"
+      - etc.
+  *)
+end
+
+(** A generator is responsible for generating pseudo-random values and provide shrinks (smaller
+    values) when a test fails. *)
```

Comment:
```suggestion
(** {2 A generator is responsible for generating pseudo-random values and provide shrinks (smaller
    values) when a test fails.} *)
```

Reply:
I'm not sure I get this proposed change?

Currently this looks like this:
![image](https://user-images.githubusercontent.com/1204125/119636390-dfe6cb80-be14-11eb-8709-3807c475c015.png)

With your proposal it would look like this:
![image](https://user-images.githubusercontent.com/1204125/119636546-07d62f00-be15-11eb-94d8-9dcdce33ea87.png)

Reply:
I don't see a difference on the left, but my bad. Seems like I need to spend more time looking at odoc's output :joy:

Reply:
Yes, there is no difference on the left, that's part of what I wanted to highlight, I should have said it explicitly :sweat_smile: